cli/command/container/stats_helpers.go

```diff
@@ -178,10 +178,14 @@ func calculateCPUPercentUnix(previousCPU, previousSystem uint64, v *types.StatsJ
 		cpuDelta = float64(v.CPUStats.CPUUsage.TotalUsage) - float64(previousCPU)
 		// calculate the change for the entire system between readings
 		systemDelta = float64(v.CPUStats.SystemUsage) - float64(previousSystem)
+		onlineCPUs = float64(v.CPUStats.OnlineCPUs)
 	)
 
+	if onlineCPUs == 0.0 {
+		onlineCPUs = float64(len(v.CPUStats.CPUUsage.PercpuUsage))
```

Comment:
Pretty hairy unfortunately. Mostly down to the first paragraph of https://github.com/docker/docker/pull/31579#issuecomment-284477640. Or we could scan sysfs and try and work it out, but that would mean opening "number of possible cpus" files in sysfs for every stats iteration and is Linux specific.